packages/cozy-harvest-lib/src/components/hooks/useAppLinkWithStoreFallback.jsx

```diff
@@ -0,0 +1,41 @@
+import { useEffect, useState } from 'react'
+import { Q, models } from 'cozy-client'
+
+const { applications: ApplicationsModel } = models
+
+const useAppLinkWithStoreFallback = (slug, client, path = '') => {
+  const [fetchStatus, setFetchStatus] = useState('loading')
+  const [isInstalled, setIsInstalled] = useState(true)
+  const [url, setURL] = useState()
+
+  useEffect(() => {
+    const load = async () => {
+      try {
+        const apps = await client.query(Q('io.cozy.apps'))
+        const appDocument = { slug }
+        const appInstalled = ApplicationsModel.isInstalled(
+          apps.data,
+          appDocument
+        )
+        setIsInstalled(!!appInstalled)
+        if (appInstalled) {
+          setURL(ApplicationsModel.getUrl(appInstalled) + path)
+        } else {
+          setURL(ApplicationsModel.getStoreURL(apps.data, appDocument))
+        }
+        setFetchStatus('loaded')
+      } catch (error) {
+        setFetchStatus('errored')
+      }
+    }
+    load()
+  }, [client, slug, path])
```

Comment:
This is not the first time that we have a `fetching` effect with an internal loading state, I remember we had one for the maintenance route, I wonder if we should have a  useFetcher hook that did that, this way we would not have to test this loading/loaded/errored logic everytime.

Reply:
This is very similar to the state that query maintain in the cozy-client state, I wonder if we could piggyback on that, I feel like we are reinventing the wheel here.